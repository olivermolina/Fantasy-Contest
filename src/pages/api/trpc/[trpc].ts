/**
 * This file contains tRPC's HTTP response handler
 */
import * as trpcNext from '@trpc/server/adapters/next';
import { createContext } from '~/server/context';
import { appRouter } from '~/server/routers/_app';

export default trpcNext.createNextApiHandler({
  router: appRouter,
  /**
   * @link https://trpc.io/docs/context
   */
  createContext,
  /**
   * @link https://trpc.io/docs/error-handling
   */
  onError({ error }) {
    if (error.code === 'INTERNAL_SERVER_ERROR') {
      // send to bug reporting
      console.error('Something went wrong', error);
    }
  },
  /**
   * Enable query batching
   */
  batching: {
    enabled: true,
  },
  /**
   * @link https://trpc.io/docs/caching#api-response-caching
   */
  responseMeta({ ctx, paths, type }) {
    // Check path if all contest queries
    const allContestPath =
      paths && paths.every((path) => path.includes('contest'));
    // checking we're doing a query request
    const isQuery = type === 'query';

    if (ctx?.res && allContestPath && isQuery) {
      // cache request for 1 day + revalidate once every second
      const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
      return {
        headers: {
          'cache-control': `s-maxage=1, stale-while-revalidate=${ONE_DAY_IN_SECONDS}`,
        },
      };
    }

    return {};
  },
});
