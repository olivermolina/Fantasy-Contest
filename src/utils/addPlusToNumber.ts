/**
 * Will add a plus sign to any positive number.
 */
export function addPlusToNumber(num: number) {
  if (num > 0) {
    return `+${num}`;
  }
  return num;
}
