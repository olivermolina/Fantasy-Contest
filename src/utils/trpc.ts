import { createTRPCNext } from '@trpc/next';
import { NextPageContext } from 'next';
import superjson from 'superjson';
import type { AppRouter } from '~/server/routers/_app';

import {
  httpBatchLink,
  httpLink,
  loggerLink,
  splitLink,
  TRPCLink,
} from '@trpc/client';
import type { inferProcedureInput, inferProcedureOutput } from '@trpc/server';
import { DefaultOptions } from '@tanstack/react-query';
import { observable } from '@trpc/server/observable';

export function getBaseUrl() {
  if (typeof window !== 'undefined') {
    return '';
  }

  // reference for vercel.com
  if (process.env.VERCEL_URL) {
    return `https://${process.env.VERCEL_URL}`;
  }

  // assume localhost
  return `http://localhost:${process.env.PORT ?? 3000}`;
}

/**
 * Extend `NextPageContext` with meta data that can be picked up by `responseMeta()` when server-side rendering
 */
export interface SSRContext extends NextPageContext {
  /**
   * Set HTTP Status code
   * @example
   * const utils = trpc.useContext();
   * if (utils.ssrContext) {
   *   utils.ssrContext.status = 404;
   * }
   */
  status?: number;
}

/**
 *  Custom link to handle authentication
 */
export const customLink: TRPCLink<AppRouter> = () => {
  // here we just got initialized in the app - this happens once per app
  // useful for storing cache for instance
  return ({ next, op }) => {
    // this is when passing the result to the next link
    // each link needs to return an observable which propagates results
    return observable((observer) => {
      return next(op).subscribe({
        next(value) {
          observer.next(value);
        },
        error(err) {
          observer.error(err);
          if (err?.data?.code === 'UNAUTHORIZED') {
            const win: Window = window;
            win.location = '/auth/login';
          }
        },
        complete() {
          observer.complete();
        },
      });
    });
  };
};

/**
 * Links to be used in both client and server
 */
export const trpcLinks = () => [
  customLink,
  loggerLink({
    enabled: (opts) =>
      process.env.NODE_ENV === 'development' ||
      (opts.direction === 'down' && opts.result instanceof Error),
  }),
  splitLink({
    condition(op) {
      // check for context property `skipBatch`
      return op.context.skipBatch === true;
    },
    // when condition is true, use normal request
    true: httpLink({
      url: `${getBaseUrl()}/api/trpc`,
    }),
    // when condition is false, use batching
    false: httpBatchLink({
      url: `${getBaseUrl()}/api/trpc`,
    }),
  }),
];

export const trpcTransformer = superjson;

/**
 * A set of strongly-typed React hooks from your `AppRouter` type signature with `createReactQueryHooks`.
 * @link https://trpc.io/docs/react#3-create-trpc-hooks
 */
export const trpcConfig: Parameters<
  typeof createTRPCNext<AppRouter, SSRContext>
>[0] = {
  config({ ctx }) {
    const commonOpts = {
      /**
       * @link https://trpc.io/docs/data-transformers
       */
      transformer: trpcTransformer,
      /**
       * @link https://trpc.io/docs/links
       */
      links: trpcLinks(),
      /**
       * @link https://react-query.tanstack.com/reference/QueryClient
       */
      queryClientConfig: {
        defaultOptions: {
          queries: {
            retry: false,
            retryOnMount: false,
            enabled: false,
            refetchInterval: 30 * 1000,
          },
        } as DefaultOptions,
      },
    };

    if (typeof window !== 'undefined') {
      // during client requests
      return {
        ...commonOpts,
        url: '/api/trpc',
      };
    }
    // during SSR below
    // optional: use SSG-caching for each rendered page (see caching section for more details)
    const ONE_DAY_SECONDS = 60 * 60 * 24;
    ctx?.res?.setHeader(
      'Cache-Control',
      `s-maxage=1, stale-while-revalidate=${ONE_DAY_SECONDS}`,
    );
    // The server needs to know your app's full url
    // On render.com you can use `http://${process.env.RENDER_INTERNAL_HOSTNAME}:${process.env.PORT}/api/trpc`
    const url = process.env.VERCEL_URL
      ? `https://${process.env.VERCEL_URL}/api/trpc`
      : 'http://localhost:3000/api/trpc';
    /**
     * If you want to use SSR, you need to use the server's full URL
     * @link https://trpc.io/docs/ssr
     */
    return {
      ...commonOpts,
      url,
      /**
       * Set custom request headers on every request from tRPC
       * @link http://localhost:3000/docs/v10/header
       * @link http://localhost:3000/docs/v10/ssr
       */
      headers() {
        if (ctx?.req) {
          // To use SSR properly, you need to forward the client's headers to the server
          // This is so you can pass through things like cookies when we're server-side rendering
          // If you're using Node 18, omit the "connection" header
          const {
            // eslint-disable-next-line @typescript-eslint/no-unused-vars
            connection: _connection,
            ...headers
          } = ctx.req.headers;
          return {
            ...headers,
            // Optional: inform server that it's an SSR request
            'x-ssr': '1',
          };
        } else {
          const { ...headers } = ctx?.req?.headers;
          return {
            ...headers,
          };
        }
      },
    };
  },
  /**
   * @link https://trpc.io/docs/ssr
   */
  ssr: false,
  // /**
  //  * Set headers or status code when doing SSR
  //  */
  // responseMeta({ ctx, clientErrors }) {
  //   if (clientErrors && clientErrors.length) {
  //     // propagate http first error from API calls
  //     return {
  //       status: clientErrors[0]!.data?.httpStatus ?? 500,
  //     };
  //   }
  //   // cache request for 1 day + revalidate once every second
  //   const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
  //   return {
  //     headers: {
  //       'cache-control': `s-maxage=1, stale-while-revalidate=${ONE_DAY_IN_SECONDS}`,
  //     },
  //   };
  // },
};

export const trpc = createTRPCNext<AppRouter, SSRContext>(trpcConfig);

/**
 * This is a helper method to infer the output of a query resolver
 * @example type HelloOutput = inferQueryOutput<'hello'>
 */
export type inferQueryOutput<
  TRouteKey extends keyof AppRouter['_def']['queries'],
> = inferProcedureOutput<AppRouter['_def']['queries'][TRouteKey]>;

export type inferQueryInput<
  TRouteKey extends keyof AppRouter['_def']['queries'],
> = inferProcedureInput<AppRouter['_def']['queries'][TRouteKey]>;

export type inferMutationOutput<
  TRouteKey extends keyof AppRouter['_def']['mutations'],
> = inferProcedureOutput<AppRouter['_def']['mutations'][TRouteKey]>;

export type inferMutationInput<
  TRouteKey extends keyof AppRouter['_def']['mutations'],
> = inferProcedureInput<AppRouter['_def']['mutations'][TRouteKey]>;
