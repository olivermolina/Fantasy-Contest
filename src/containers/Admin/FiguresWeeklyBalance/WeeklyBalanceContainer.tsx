import React, { useMemo } from 'react';
import { trpc } from '~/utils/trpc';
import TableFilter from '~/components/Admin/Figures/WeeklyBalance/TableFilter';
import _ from 'lodash';
import BalanceTableByAgent from '~/components/Admin/Figures/WeeklyBalance/BalanceTableByAgent';
import {
  Dialog,
  DialogTitle,
  IconButton,
  Skeleton,
  Typography,
} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import { DateRangeInterface } from '~/components/Picks/PickDatePickerRange';
import { PickStatus } from '~/constants/PickStatus';
import { DayOfWeek } from '~/constants/DayOfWeek';
import { BalanceDateRage } from '~/server/routers/admin/figures/weeklyBalances/playerWeeklyBalance';
import BalanceSummaryTable from '~/components/Admin/Figures/WeeklyBalance/BalanceSummaryTable';
import PicksContainer from '~/containers/PicksContainer/PicksContainer';
import type { IPlayerWeeklyBalance } from '~/server/routers/admin/figures/weeklyBalances/weeklyBalances';
import dayjs from 'dayjs';
import weekday from 'dayjs/plugin/weekday';
import updateLocale from 'dayjs/plugin/updateLocale';
import utc from 'dayjs/plugin/utc';

dayjs.extend(utc);
dayjs.extend(updateLocale);
dayjs.extend(weekday);
dayjs.updateLocale('en', {
  weekStart: 1,
});

const WeeklyBalanceContainer = () => {
  const handleClose = () => {
    setOpen(false);
  };
  const [showTabTitle, setShowTabTitle] = React.useState(false);
  const [open, setOpen] = React.useState(false);
  const [date, setDate] = React.useState(new Date());
  const [viewInactive, setViewInactive] = React.useState(false);
  const [includeEntryFee, setIncludeEntryFee] = React.useState(false);
  const { isLoading, data } = trpc.admin.weeklyBalances.useQuery({
    date: dayjs(date).format('YYYY-MM-DD'),
    includeInactive: viewInactive,
    includeEntryFee,
  });
  const [globalFilter, setGlobalFilter] = React.useState('');
  const [dateRangeValue, setDateRangeValue] = React.useState<
    DateRangeInterface | undefined
  >(undefined);
  const [selectedPlayer, setSelectedPlayer] =
    React.useState<IPlayerWeeklyBalance | null>(null);
  const [selectedTabStatus, setSelectedTabStatus] = React.useState<PickStatus>(
    PickStatus.SETTLED,
  );

  const agents = useMemo(
    () =>
      _.uniqBy(
        data?.weeklyBalances?.map((player) => player.agent),
        'id',
      ),
    [data],
  );

  const handleSetDateFilter = (
    dateRange: BalanceDateRage,
    dayOfWeek?: DayOfWeek,
  ) => {
    const startDateFilter = dayjs(dateRange?.from)
      .weekday(dayOfWeek || DayOfWeek.MONDAY)
      .format('YYYY-MM-DD');
    const endDateFilter = dayjs(dateRange?.from)
      .weekday(dayOfWeek !== undefined ? dayOfWeek : DayOfWeek.SUNDAY)
      .format('YYYY-MM-DD');

    setDateRangeValue({
      startDate: dayjs(startDateFilter),
      endDate: dayjs(endDateFilter),
    });
  };

  const setIncludeEntryFeeOnClick = (includeEntryFee: boolean) => {
    setIncludeEntryFee(includeEntryFee);
  };

  return (
    <div className={'flex flex-col gap-2'}>
      <h2 className={'text-xl font-bold'}>Weekly Balance</h2>
      <TableFilter
        setDate={setDate}
        globalFilter={globalFilter}
        setGlobalFilter={setGlobalFilter}
        viewInactive={viewInactive}
        setViewInactive={setViewInactive}
        includeEntryFee={includeEntryFee}
        setIncludeEntryFeeOnClick={setIncludeEntryFeeOnClick}
      />
      <div className="flex flex-col justify-center items-center">
        <p>- Weekly Balance -</p>
        {isLoading ? (
          <Skeleton sx={{ width: 200 }} />
        ) : (
          <p>
            {dayjs(data?.dateRange?.from).format('MM/DD/YYYY')} to{' '}
            {dayjs(data?.dateRange?.to).format('MM/DD/YYYY')}
          </p>
        )}
      </div>
      {isLoading ? (
        <Skeleton variant={'rectangular'} sx={{ height: 200 }} />
      ) : (
        _.orderBy(agents, (agent) => agent?.id || '', ['desc']).map((agent) => (
          <BalanceTableByAgent
            key={agent?.id}
            data={
              data?.weeklyBalances.filter(
                (player) =>
                  player?.agent?.id === agent?.id &&
                  player.isActive === !viewInactive,
              ) || []
            }
            agent={agent}
            globalFilter={globalFilter}
            setGlobalFilter={setGlobalFilter}
            setOpen={setOpen}
            handleSetDateFilter={handleSetDateFilter}
            setSelectedPlayer={setSelectedPlayer}
            dateRange={
              data?.dateRange || { from: date.toString(), to: date.toString() }
            }
            setSelectedTabStatus={setSelectedTabStatus}
            setShowTabTitle={setShowTabTitle}
          />
        ))
      )}

      {!isLoading && (
        <BalanceSummaryTable
          data={data?.weeklyBalances || []}
          viewInactive={viewInactive}
          activeCount={data?.activeCount || 0}
          inactiveCount={data?.inactiveCount || 0}
        />
      )}

      <Dialog
        open={open}
        fullScreen
        fullWidth
        sx={{ top: 65 }}
        onClose={handleClose}
        className={'bg-secondary'}
      >
        <DialogTitle sx={{ m: 0, p: 2 }}>
          <Typography variant={'inherit'} sx={{ textAlign: 'center' }}>
            <p>{selectedPlayer?.username}</p>
            <p>
              -{' '}
              {selectedTabStatus === PickStatus.PENDING ? 'PENDING' : 'SETTLED'}{' '}
              PICKS -
            </p>
            {selectedTabStatus === PickStatus.SETTLED && (
              <p>
                {dayjs(dateRangeValue?.startDate).format('MM/DD/YYYY')} to{' '}
                {dayjs(dateRangeValue?.endDate).format('MM/DD/YYYY')}
              </p>
            )}
          </Typography>
          <IconButton
            aria-label="close"
            onClick={handleClose}
            sx={{
              position: 'absolute',
              right: 8,
              top: 8,
              color: (theme) => theme.palette.grey[500],
            }}
          >
            <CloseIcon />
          </IconButton>
        </DialogTitle>
        <div className={'bg-secondary'}>
          <PicksContainer
            userId={selectedPlayer?.id || ''}
            defaultDateRange={
              selectedTabStatus === PickStatus.SETTLED
                ? dateRangeValue
                : undefined
            }
            showBalanceSummary={false}
            showDateFilters={false}
            showTabTitle={showTabTitle}
            selectedTabStatus={selectedTabStatus}
            isAdminView
          />
        </div>
      </Dialog>
    </div>
  );
};

export default WeeklyBalanceContainer;
