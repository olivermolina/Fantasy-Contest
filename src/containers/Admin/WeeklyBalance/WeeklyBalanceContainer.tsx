import React, { useMemo } from 'react';
import { trpc } from '~/utils/trpc';
import TableFilter from '~/components/Admin/Figures/WeeklyBalance/TableFilter';
import _ from 'lodash';
import BalanceTableByAgent from '~/components/Admin/Figures/WeeklyBalance/BalanceTableByAgent';
import {
  Dialog,
  DialogTitle,
  IconButton,
  Skeleton,
  Typography,
} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import { DateRangeInterface } from '~/components/Picks/PickDatePickerRange';
import { PickStatus } from '~/constants/PickStatus';
import { DayOfWeek } from '~/constants/DayOfWeek';
import dayjs from 'dayjs';
import weekday from 'dayjs/plugin/weekday';
import updateLocale from 'dayjs/plugin/updateLocale';
import { WeeklyBalanceDateRage } from '~/server/routers/admin/figures/weeklyBalances/playerWeeklyBalance';
import BalanceSummaryTable from '~/components/Admin/Figures/WeeklyBalance/BalanceSummaryTable';
import PicksContainer from '~/containers/PicksContainer/PicksContainer';

dayjs.extend(updateLocale);
dayjs.updateLocale('en', {
  weekStart: 1,
});

dayjs.extend(weekday);

const WeeklyBalanceContainer = () => {
  const handleClose = () => {
    setOpen(false);
  };
  const [showTabTitle, setShowTabTitle] = React.useState(false);
  const [open, setOpen] = React.useState(false);
  const [date, setDate] = React.useState(new Date());
  const { isLoading, data } = trpc.admin.weeklyBalances.useQuery({
    date: dayjs(date).format('YYYY-MM-DD'),
  });
  const [globalFilter, setGlobalFilter] = React.useState('');
  const [dateRangeValue, setDateRangeValue] = React.useState<
    DateRangeInterface | undefined
  >(undefined);
  const [userId, setUserId] = React.useState('');
  const [selectedTabStatus, setSelectedTabStatus] = React.useState<PickStatus>(
    PickStatus.SETTLED,
  );

  const [viewInactive, setViewInactive] = React.useState(false);

  const agents = useMemo(
    () =>
      _.uniqBy(
        data?.weeklyBalances?.map((player) => player.agent),
        'id',
      ),
    [data],
  );

  const handleSetDateFilter = (
    dateRange: WeeklyBalanceDateRage,
    dayOfWeek?: DayOfWeek,
  ) => {
    const startDateFilter = dayjs(dateRange?.from)
      .weekday(dayOfWeek || DayOfWeek.MONDAY)
      .format('YYYY-MM-DD');
    const endDateFilter = dayjs(dateRange?.from)
      .weekday(dayOfWeek !== undefined ? dayOfWeek : DayOfWeek.SUNDAY)
      .format('YYYY-MM-DD');

    setDateRangeValue({
      startDate: dayjs(startDateFilter),
      endDate: dayjs(endDateFilter),
    });
  };

  return (
    <div className={'flex flex-col gap-2'}>
      <h2 className={'text-xl font-bold'}>Weekly Balance</h2>
      <TableFilter
        setDate={setDate}
        globalFilter={globalFilter}
        setGlobalFilter={setGlobalFilter}
        viewInactive={viewInactive}
        setViewInactive={setViewInactive}
      />
      <div className="flex flex-col justify-center items-center">
        <p>- Weekly Balance -</p>
        {isLoading ? (
          <Skeleton sx={{ width: 200 }} />
        ) : (
          <p>
            {dayjs(data?.dateRange?.from).format('MM/DD/YYYY')} to{' '}
            {dayjs(data?.dateRange?.to).format('MM/DD/YYYY')}
          </p>
        )}
      </div>
      {isLoading ? (
        <Skeleton variant={'rectangular'} sx={{ height: 200 }} />
      ) : (
        _.orderBy(agents, (agent) => agent?.id || '', ['desc']).map((agent) => (
          <BalanceTableByAgent
            key={agent?.id}
            data={
              data?.weeklyBalances.filter(
                (player) =>
                  player?.agent?.id === agent?.id &&
                  player.isActive === !viewInactive,
              ) || []
            }
            agent={agent}
            globalFilter={globalFilter}
            setGlobalFilter={setGlobalFilter}
            setOpen={setOpen}
            handleSetDateFilter={handleSetDateFilter}
            setUserId={setUserId}
            dateRange={
              data?.dateRange || { from: date.toString(), to: date.toString() }
            }
            setSelectedTabStatus={setSelectedTabStatus}
            setShowTabTitle={setShowTabTitle}
          />
        ))
      )}

      {!isLoading && (
        <BalanceSummaryTable
          data={data?.weeklyBalances || []}
          viewInactive={viewInactive}
        />
      )}

      <Dialog
        open={open}
        fullScreen
        fullWidth
        sx={{ top: 65 }}
        onClose={handleClose}
        className={'bg-secondary'}
      >
        <DialogTitle sx={{ m: 0, p: 2 }}>
          <Typography variant={'inherit'} sx={{ textAlign: 'center' }}>
            <p>
              -{' '}
              {selectedTabStatus === PickStatus.PENDING ? 'PENDING' : 'SETTLED'}{' '}
              PICKS -
            </p>
            {selectedTabStatus === PickStatus.SETTLED && (
              <p>
                {dayjs(dateRangeValue?.startDate).format('MM/DD/YYYY')} to{' '}
                {dayjs(dateRangeValue?.endDate).format('MM/DD/YYYY')}
              </p>
            )}
          </Typography>
          <IconButton
            aria-label="close"
            onClick={handleClose}
            sx={{
              position: 'absolute',
              right: 8,
              top: 8,
              color: (theme) => theme.palette.grey[500],
            }}
          >
            <CloseIcon />
          </IconButton>
        </DialogTitle>
        <div className={'bg-secondary'}>
          <PicksContainer
            userId={userId}
            defaultDateRange={
              selectedTabStatus === PickStatus.SETTLED
                ? dateRangeValue
                : undefined
            }
            showBalanceSummary={false}
            showDateFilters={false}
            showTabTitle={showTabTitle}
            selectedTabStatus={selectedTabStatus}
            isAdminView
          />
        </div>
      </Dialog>
    </div>
  );
};

export default WeeklyBalanceContainer;
