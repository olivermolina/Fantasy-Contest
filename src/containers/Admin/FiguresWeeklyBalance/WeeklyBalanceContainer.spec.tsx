import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { trpc } from '~/utils/trpc';
import WeeklyBalanceContainer from '~/containers/Admin/FiguresWeeklyBalance/WeeklyBalanceContainer';

jest.mock('~/utils/trpc', () => ({
  trpc: {
    admin: {
      weeklyBalances: {
        useQuery: jest.fn().mockReturnValue({
          isLoading: true,
          data: {
            dateRange: {
              from: new Date('2023-01-30'),
              to: new Date('2023-02-05'),
            },
            weeklyBalances: [],
          },
        }),
      },
    },
  },
}));

describe('WeeklyBalanceContainer', () => {
  it('should render correctly', () => {
    (trpc.admin.weeklyBalances.useQuery as jest.Mock).mockReturnValue({
      isLoading: true,
      data: [],
    });

    const { container } = render(<WeeklyBalanceContainer />);
    expect(container).toMatchSnapshot();
  });
});
