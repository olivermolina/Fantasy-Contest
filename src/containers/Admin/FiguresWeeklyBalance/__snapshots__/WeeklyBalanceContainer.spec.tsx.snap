// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyBalanceContainer should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <h2
      class="text-xl font-bold"
    >
      Weekly Balance
    </h2>
    <div
      class="flex flex-col gap-2 py-2 w-full"
    >
      <div
        class="flex flex-col md:flex-row gap-2 w-full"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            Day Week Date
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="Choose date, selected date is 28 Dec, 2020"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-readOnly MuiInputBase-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              inputmode="text"
              placeholder="MM/DD/YYYY"
              readonly=""
              type="text"
              value="12/28/2020"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Day Week Date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="w-full lg:max-w-sm"
        >
          <div
            class="relative grow"
          >
            <div
              class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5 text-gray-500 dark:text-gray-400"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                />
              </svg>
            </div>
            <input
              class="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-2 focus:ring-blue-500 focus:border-blue-500 focus:outline-none"
              placeholder="Search"
              required=""
              type="search"
              value=""
            />
          </div>
        </div>
        <div
          class="flex flex-row gap-2 w-full"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                aria-label="controlled"
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              View inactive only
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                aria-label="controlled"
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Include Entry Fee
            </span>
          </label>
        </div>
      </div>
      <div
        class="flex flex-row-reverse gap-2 justify-end"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          This week
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Last week
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          2 weeks ago
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          3 weeks ago
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          4 weeks ago
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          5 weeks ago
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          6 weeks ago
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          7 weeks ago
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          8 weeks ago
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="flex flex-col justify-center items-center"
    >
      <p>
        - Weekly Balance -
      </p>
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1bjvsl0-MuiSkeleton-root"
      />
    </div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-rofme1-MuiSkeleton-root"
    />
  </div>
</div>
`;
